src/streamers/spotify.streamer.ts

```diff
@@ -37,14 +37,18 @@ export class SpotifyStreamer implements IStreamer {
   private clientSecret: string;
   private refreshToken: string;
   private storer: IStorer;
+  private useCache: boolean;
+  private cacheDuration: number;
 
-  constructor(storer: IStorer, args: SpotifyStreamerArgs) {
+  constructor(storer: IStorer, args: SpotifyStreamerArgs & IStreamerCacheOpts) {
     this.storer = storer;
     SpotifyStreamerArgsSchema.parse(args);
 
     this.clientId = args.clientId;
     this.clientSecret = args.clientSecret;
     this.refreshToken = args.refreshToken;
+    this.useCache = args.useCache;
+    this.cacheDuration = args.cacheDuration;
```

Comment:
I think it's better to set these options in the streamer class so that all the methods can have access to it instead of passing it around.